Makefile

```diff
@@ -1,36 +1,53 @@
-GO_BUILD_FLAGS = 
+GO_BUILD_FLAGS =
 DOCKER_PLUGINS = docker-generate docker-nv2
 COMMANDS       = nv2 $(DOCKER_PLUGINS)
+GOPATH := $(shell go env GOPATH)
 
 define BUILD_BINARY =
 	go build $(GO_BUILD_FLAGS) -o $@ ./$<
 endef
 
+.PHONY: help
+
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-25s\033[0m %s\n", $$1, $$2}'
+
 .PHONY: all
 all: build
 
 bin/%: cmd/%
 	$(BUILD_BINARY)
 
+.PHONY: download
+download: ## download dependencies via go mod
+	go mod download
+
 .PHONY: build
-build: $(addprefix bin/,$(COMMANDS))
+build: $(addprefix bin/,$(COMMANDS)) ## builds binaries
 
 .PHONY: clean
 clean:
 	git status --ignored --short | grep '^!! ' | sed 's/!! //' | xargs rm -rf
 
-.PHONY: check-encoding
-check-encoding:
+.PHONY: check-line-endings
+check-line-endings: ## check line endings
 	! find cmd pkg internal -name "*.go" -type f -exec file "{}" ";" | grep CRLF
 
-.PHONY: fix-encoding
-fix-encoding:
+.PHONY: fix-line-endings
+fix-line-endings: ## fix line endings
 	find cmd pkg internal -type f -name "*.go" -exec sed -i -e "s/\r//g" {} +
 
 .PHONY: vendor
-vendor:
+vendor: ## vendores the go modules
 	GO111MODULE=on go mod vendor
 
 .PHONY: install-docker-plugins
-install-docker-plugins: $(addprefix bin/,$(DOCKER_PLUGINS))
+install-docker-plugins: $(addprefix bin/,$(DOCKER_PLUGINS)) ## installs the docker plugins
 	cp $(addprefix bin/,$(DOCKER_PLUGINS)) ~/.docker/cli-plugins/
+
+lint: ## use goimports to lint
+	$(GOPATH)/bin/goimports -d -e -local github.com/notaryproject/nv2 $$(go list -f {{.Dir}} ./...)
```

Comment:
Let's try not embed any go tools in our repo.

Reply:
This tool is very commonly used in vscode to format the file and organize imports.

This tool enables to verify this using the same settings during CI/CD to maintain a consistent code style and usage of imports.

Reply:
It is common in vscode to use `goimports` to  organize imports but it also gives many options like `goimports`, `goreturns`, or `gofumports`. Furthermore, those tools are usually installed outside of this repo.

For CI, we should use `go fmt` since `goimports` is a superset of `go fmt` with the ability to manage imports. If there is a missing package in the source code, we must treat it as an error and not auto-resolved by `goimports`.

Reply:
The build will fail anyhow. gomimports will not install it in CI.

Reply:
I removed the linting from this PR as this is a point of discussion. can create a new PR once we come to common understanding.